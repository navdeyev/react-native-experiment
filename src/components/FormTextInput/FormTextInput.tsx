import *as React from 'react';
import {Text, TextInput, View} from 'react-native';
import {EventOrValueHandler} from 'redux-form';
import styled from 'styled-components';

const Wrapper = styled(View)`margin: 0 0 6px 0;`;
const InputLabel = styled(Text)`font-weight: bold;`;
const StyledTextInput = styled(TextInput)`padding: 0 0 2px 0;`;
const InputError = styled(Text)`color: #ff0000`;

interface WrappedInput {
    value: any;
    onChange: EventOrValueHandler<React.ChangeEvent<any>>
}

interface WrappedInputMeta {
    error?: any;
}

export interface WrappedInputField {
    label?: string;
    input: WrappedInput;
    meta: WrappedInputMeta
}

const FormTextInput = (props: WrappedInputField) => {
    const label = props.label;
    const {onChange, value} = props.input;
    const {error} = props.meta;
    return (
        <Wrapper>
            {label ? <InputLabel testID="form-text-input-label">{label}</InputLabel> : null}
            <StyledTextInput onChangeText={onChange} value={value}/>
            {error ? <InputError testID="form-text-input-error">{error}</InputError> : null}
        </Wrapper>
    );
};

export default FormTextInput;