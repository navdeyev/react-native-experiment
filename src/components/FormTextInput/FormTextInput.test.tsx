import * as React from 'react';
import {shallow} from 'enzyme';

import FormTextInput, {WrappedInputField} from './FormTextInput';

describe('FormTextInput', () => {

    let props: WrappedInputField;
    beforeEach(() => {
        props = {
            input: {value: '', onChange: jest.fn()},
            meta: {}
        };
    });

    it('renders the TextInput', () => {
        const rendered = shallow(<FormTextInput {...props}/>);
        expect(rendered.find('[testID="form-text-input-label"]').exists()).toBe(false);
        expect(rendered.find('[testID="form-text-input-error"]').exists()).toBe(false);
        expect(rendered.getElement()).toMatchSnapshot();
    });

    it('renders the TextInput', () => {
        const propsWithLabelAndError = {
            ...props,
            label: 'Field label',
            meta: { error: 'Please add a value' }
        };
        const rendered = shallow(<FormTextInput {...propsWithLabelAndError}/>);
        expect(rendered.find('[testID="form-text-input-label"]').exists()).toBe(true);
        expect(rendered.find('[testID="form-text-input-error"]').exists()).toBe(true);
        expect(rendered.getElement()).toMatchSnapshot();
    });

});