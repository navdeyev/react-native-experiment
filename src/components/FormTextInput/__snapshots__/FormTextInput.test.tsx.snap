// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormTextInput renders the TextInput 1`] = `
<Styled(Component)>
  <Styled(TextInput)
    onChangeText={[MockFunction]}
    value=""
  />
</Styled(Component)>
`;

exports[`FormTextInput renders the TextInput 2`] = `
<Styled(Component)>
  <Styled(Text)
    testID="form-text-input-label"
  >
    Field label
  </Styled(Text)>
  <Styled(TextInput)
    onChangeText={[MockFunction]}
    value=""
  />
  <Styled(Text)
    testID="form-text-input-error"
  >
    Please add a value
  </Styled(Text)>
</Styled(Component)>
`;
